### PR TITLE
#759 google ecommerce

### LINKED ISSUE #759: Broken google ecommerce integration
https://github.com/fidals/shopelectro/issues/706

20 feb 2019 we broke google analytics. Resurrect it.
Check if tests for ecommerce are working.

(under mediaprime acc)
https://analytics.google.com/analytics/web/?authuser=0#/report/conversions-ecommerce-overview/a42239144w71929328p74216659/_u.date00=20190201&_u.date01=20190305/

image

### PR DESCRIPTION
**Pull request by artemiy312**
Closes #759

### CODE REVIEW

File: front/js/shared/tracking.es6 [hunk 1]
```diff
@@ -78,6 +67,44 @@
       });
   }
 
+  // @todo #759:30m Move PublishedGATracker to a separate file.
+
+  class PublishedGATracker {
```
Comment (duker33) [inline front/js/shared/tracking.es6]: provide doc plz why it's published, but not common `GATracker`

Comment (duker33) [reply front/js/shared/tracking.es6]: publish is not the most good name for classes purpose. `LimitedGATracher` or `SafeGATracker` might be more good. Up to you

File: front/js/shared/tracking.es6 [hunk 2]
```diff
@@ -78,6 +67,44 @@
...
+    purchase(productsData, txData) {
+      const publishOnce = () => {
+          // Publish only once
+          if (this.published) return;
+
+          // Load ecommerce plugin for gaTracker
+          ga('require', 'ecommerce');  // Ignore ESLintBear (block-scoped-var)
+          const tracker = new GATracker(ga, 'ecommerce');  // Ignore ESLintBear (block-scoped-var)
+
+          tracker.purchase(productsData, txData);
```
Comment (duker33) [inline front/js/shared/tracking.es6]: hope data is consistent with gAnalytics api docs. Looks too simple and suspicious =)

File: front/js/shared/tracking.es6 [hunk 3]
```diff
@@ -78,6 +67,44 @@
...
   }
 
+  // @todo #759:30m Move PublishedGATracker to a separate file.
+
+  class PublishedGATracker {
+    constructor() {
+      this.published = false;
+    }
+
+    purchase(productsData, txData) {
```
Comment (duker33) [inline front/js/shared/tracking.es6]: maybe it's possible to add tests for ecommerce behaviour.
We are already testing Ya.Metrika at SE: `shopelectro.tests.tests_selenium.YandexMetrika`

File: front/js/shared/tracking.es6 (hunk 1 above)
Comment (artemiy312) [reply front/js/shared/tracking.es6]: I believe `LoadedGATracker` would be the best name

**Pull request merged**